packages/mobile/src/screens/chat-screen/ChatUserListScreen.tsx

```diff
@@ -177,6 +113,17 @@ export const ChatUserListScreen = (props: ChatUserListScreenProps) => {
   )
   const isLoading = hasQuery && status === Status.LOADING
 
+  useEffect(() => {
+    dispatch(fetchBlockees())
+    dispatch(fetchBlockers())
+  }, [dispatch])
+
+  useEffect(() => {
+    if (userIds.length > 0) {
+      dispatch(fetchPermissions({ userIds }))
+    }
+  }, [dispatch, userIds])
```

Comment:
Do you think it would make sense to push this logic into the sagas? i.e.:
* The loading state for this component is based on waiting for the user list, list of blockees, and list of blockers. We could fetch them all in parallel in the same loading saga and only switch the state once we've finished all three
* Can/should we move fetching permissions to the saga that populates user ids? If we fork it there, it's not waiting for a re-render and effect hook loop before fetching the permissions.

Reply:
Hm.. I don't think we want to couple the user list fetching with permissions fetching. Right now this is the only place we use this `UserList` sort of component but I'd like to make it generic.

Reply:
Another possibility is to pull this out into a shared hook if this logic applies in web as well - cc @rickyrombo

Reply:
Oh okay. I thought the saga for fetching the user list was specifically for chats. So I would have thought it could fetch the users and decorate them with their status. Or fetch both users and block information and combine.
Or, fetch users and proactively start fetching blocks.
Anyway a hook composing the two also makes sense.

Reply:
I still need to adopt this actually in web! I'll work on extracting a hook for this